pkg/resources/mutation/cluster/provisioningcluster.go

```diff
@@ -53,10 +97,258 @@ func (m *Mutator) Admit(request *admission.Request) (*admissionv1.AdmissionRespo
 	if err != nil {
 		return nil, err
 	}
-	response := &admissionv1.AdmissionResponse{}
-	if err := mutation.SetCreatorIDAnnotation(request, response, request.Object, cluster.DeepCopy()); err != nil {
-		return nil, fmt.Errorf("failed to set creatorIDAnnotation: %w", err)
+
+	if request.Operation == admissionv1.Create {
+		// Set Annotation on the cluster
+		annotations := cluster.GetAnnotations()
+		if annotations == nil {
+			annotations = map[string]string{}
+		}
+		annotations[auth.CreatorIDAnn] = request.UserInfo.Username
+		cluster.SetAnnotations(annotations)
 	}
+
+	response, err := m.handlePSACT(request, cluster)
+	if err != nil {
+		return nil, err
+	}
+	if response.Result != nil {
+		return response, nil
+	}
+
 	response.Allowed = true
+	if err = patch.CreatePatch(request.Object.Raw, cluster, response); err != nil {
+		return nil, fmt.Errorf("failed to create patch: %w", err)
+	}
 	return response, nil
 }
+
+// handlePSACT updates the cluster and an underlying secret to support PSACT.
+// If a PSACT is set in the cluster, handlePSACT generates an admission configuration file, put it into a secret,
+// updates the cluster's spec to mount the secret to the control plane nodes, and configures kube-apisever to use the admission configuration file;
+// If a PSACT is unset in the cluster, handlePSACT does the cleanup for both the secret and cluster's spec.
+func (m *ProvisioningClusterMutator) handlePSACT(request *admission.Request, cluster *v1.Cluster) (*admissionv1.AdmissionResponse, error) {
+	if cluster.Name == "local" || cluster.Spec.RKEConfig == nil {
+		return admission.ResponseAllowed(), nil
+	}
+	parsedVersion, err := psa.GetClusterVersion(cluster.Spec.KubernetesVersion)
+	if err != nil {
+		return nil, fmt.Errorf("[provsioning cluster mutator] failed to parse cluster version: %w", err)
+	}
+	if parsedRangeLessThan123(parsedVersion) {
+		return admission.ResponseAllowed(), nil
+	}
+
+	secretName := fmt.Sprintf(SecretName, cluster.Name)
+	mountPath := fmt.Sprintf(MountPath, GetRuntime(cluster.Spec.KubernetesVersion))
+	templateName := cluster.Spec.DefaultPodSecurityAdmissionConfigurationTemplateName
+
+	switch request.Operation {
+	case admissionv1.Delete:
+		err := m.secret.Delete(cluster.Namespace, secretName, &metav1.DeleteOptions{})
+		if err != nil && !apierrors.IsNotFound(err) {
+			return nil, fmt.Errorf("[provsioning cluster mutator] failed to delete the secret: %w", err)
+		}
+	case admissionv1.Create, admissionv1.Update:
+		if cluster.DeletionTimestamp != nil {
+			return admission.ResponseAllowed(), nil
+		}
+		if templateName == "" {
+			err := m.secret.Delete(cluster.Namespace, secretName, &metav1.DeleteOptions{})
+			if err != nil && !apierrors.IsNotFound(err) {
+				return nil, fmt.Errorf("[provsioning cluster mutator] failed to delete the secret: %w", err)
+			}
+			// drop relevant fields if they exist in the cluster
```

Comment:
Shouldn't the logic for dropping relevant fields from the cluster management object also be under the admission.v1 Delete case? If those fields still exist, and you're just deleting the secret they would be incorrectly left behind on the object.. but lmk if I just missed something because I haven't had enough Monday coffee yet

In general, I think a few more `Now, handle the case of ` comments in this section like the one on line 169 would be helpful